scripts/babel-relay-plugin/src/GraphQLPrinter.js

```diff
@@ -290,6 +290,14 @@ function printFieldsAndFragments(
         fragments.push(printQueryFragment(selection, options));
       } else if (selection.kind === kinds.FIELD) {
         fields.push(selection);
+
+        if (
+          isConnectionType(type) &&
```

Comment:
On closer inspection, I don't think this is possible. For a fragment, I don't actually have access to the field, just the type, so I can't make these assertions because I don't have access to the field args when looking at just the fragment: https://github.com/facebook/relay/blob/d2c821efb6f9620c7e3597ad92b3e1e1a838829e/scripts/babel-relay-plugin/src/GraphQLPrinter.js#L793-L795.

What I can do is to lift these into a separate function: https://github.com/facebook/relay/blob/d2c821efb6f9620c7e3597ad92b3e1e1a838829e/scripts/babel-relay-plugin/src/GraphQLPrinter.js#L796-L826 and use that instead. Does that sound like a workable approach?